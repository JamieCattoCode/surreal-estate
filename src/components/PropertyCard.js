import React from 'react';
import PropTypes from 'prop-types';
import '../styles/property-card.css';

const PropertyCard = ({
  title, type, bedrooms, bathrooms, price, city, email,
}) => (
  <div className="property-card">
    <div className="property-card__title">
      {title}
    </div>
    <div className="property-card__type-city">
      {type}
      {' '}
      -
      {' '}
      {city}
    </div>
    <div className="property-card__bedrooms">
      {bedrooms}
      {' '}
      bedrooms
    </div>
    <div className="property-card__bathrooms">
      {bathrooms}
      {' '}
      bathrooms
    </div>
    <div className="property-card__price">
      Â£
      {price}
    </div>
    <div className="property-card__email">
      {email}
    </div>
  </div>
);

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bedrooms: PropTypes.number.isRequired,
  bathrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};

export default PropertyCard;
