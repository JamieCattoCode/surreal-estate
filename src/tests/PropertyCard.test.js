import React from 'react';
import { render } from '@testing-library/react';
import PropertyCard from '../components/PropertyCard';

describe('PropertyCard component', () => {
  const validProps = {
    title: '2 bed flat',
    type: 'Apartment',
    bedrooms: 2,
    bathrooms: 3,
    price: 150000,
    city: 'Manchester',
    email: 'jamie@surrealestate.com',
  };
  const { asFragment } = render(
    <PropertyCard
      title={validProps.title}
      type={validProps.type}
      bedrooms={validProps.bedrooms}
      bathrooms={validProps.bathrooms}
      price={validProps.price}
      city={validProps.city}
      email={validProps.email}
    />,
  );

  it('renders a property card correctly', () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the correct value for props', () => {
    const { getByText } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bedrooms={validProps.bedrooms}
        bathrooms={validProps.bathrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />,
    );
    expect(getByText(/2 bed flat/i)).toHaveClass('property-card__title');
    expect(getByText(/apartment/i)).toHaveClass('property-card__type-city');
    expect(getByText(/manchester/i)).toHaveClass('property-card__type-city');
    expect(getByText(/2 bedrooms/i)).toHaveClass('property-card__bedrooms');
    expect(getByText(/3 bathrooms/i)).toHaveClass('property-card__bathrooms');
    expect(getByText(/Â£150000/i)).toHaveClass('property-card__price');
  });
});
