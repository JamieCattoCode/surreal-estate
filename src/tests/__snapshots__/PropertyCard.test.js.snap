// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard component renders a property card correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="property-card__title"
    >
      2 bed flat
    </div>
    <div
      class="property-card__type-city"
    >
      Apartment -Manchester
    </div>
    <div
      class="property-card__bedrooms"
    >
      2 bedrooms
    </div>
    <div
      class="property-card__bathrooms"
    >
      3 bathrooms
    </div>
    <div
      class="property-card__price"
    >
      Â£150000
    </div>
  </div>
</DocumentFragment>
`;
